src/main/java/org/opensha/sha/earthquake/faultSysSolution/inversion/constraints/impl/MFDUncertaintyWeightedInversionConstraint.java

```diff
@@ -0,0 +1,111 @@
+package org.opensha.sha.earthquake.faultSysSolution.inversion.constraints.impl;
+
+import java.util.HashSet;
+import java.util.List;
+
+import org.opensha.commons.data.function.EvenlyDiscretizedFunc;
+import org.opensha.sha.earthquake.faultSysSolution.FaultSystemRupSet;
+import org.opensha.sha.earthquake.faultSysSolution.inversion.constraints.InversionConstraint;
+import org.opensha.sha.magdist.IncrementalMagFreqDist;
+
+import com.google.common.base.Preconditions;
+
+import cern.colt.matrix.tdouble.DoubleMatrix2D;
+import scratch.UCERF3.utils.MFD_InversionConstraint;
+import scratch.UCERF3.utils.MFD_WeightedInversionConstraint;
+
+/**
+ * Constrain the solution to match the given MFD Weighted constraints. 
+ * 
+ * @author chrisbc
+ *
+ */
+public class MFDUncertaintyWeightedInversionConstraint extends InversionConstraint {
+	
+	public static final String NAME = "MFD UncertaintyWeighted";
+	public static final String SHORT_NAME = "MFDUncertaintyWeighted";
+	
+	private FaultSystemRupSet rupSet;
+	private double weight;
+	private List<MFD_WeightedInversionConstraint> mfdWeightedConstraints;
+	private HashSet<Integer> excludeRupIndexes;
+
+	public MFDUncertaintyWeightedInversionConstraint(FaultSystemRupSet rupSet, double weight,
+			List<MFD_WeightedInversionConstraint> mfdWeightedConstraints) {
+		this.rupSet = rupSet;
+		this.weight = weight;
+		this.mfdWeightedConstraints = mfdWeightedConstraints;
+	}
+
+	@Override
+	public String getShortName() {
+		return SHORT_NAME;
+	}
+
+	@Override
+	public String getName() {
+		return NAME;
+	}
+
+	@Override
+	public int getNumRows() {
```

Comment:
looks equal to the implementation in `MFDEqualityInversionConstraint` apart from the constraints list name. Maybe pull this out into a shared method?

Reply:
@voj And also the same as MFDInequality... I agree this might be refactored to be more DRY, but I'll let Kevin review as-is first.

Reply:
Looks fine to me, but I agree that this method could be moved to a shared static method. I went ahead an added one in 59d418d that you can use. Otherwise looks good, with the caveat that we may end up renaming these things down the line if we go with a different uncertainty scheme

Reply:
done